generators/Exercises/Generators/AffineCipher.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Exercism.CSharp.Output;
+
+namespace Exercism.CSharp.Exercises.Generators
+{
+    public class AffineCipher : GeneratorExercise
```

Comment:
There is quite a bit of customization going on here, and I think we can simplify things quite a bit. Let me explain!

Looking at the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/affine-cipher/canonical-data.json), the input data is straightforward, except for the `"key"` property, which is a nested property. You can work around this using the following code in the `UpdateTestMethod` method:

```csharp
testMethod.Input["a"] = testMethod.Input["key"]["a"];
testMethod.Input["b"] = testMethod.Input["key"]["b"];
testMethod.InputParameters = new[] { "phrase", "a", "b" };
```

The second bit is dealing with exception, which is also easy to fix:

```csharp
if (!(testMethod.Expected is string))
     testMethod.ExceptionThrown = typeof(ArgumentException);
```

With just these 5 lines of code, the generate becomes a lot simpler and still generates a nice test file. The only change you have to make in your example and stub files is make them use a static class with static methods.

Reply:
Generator is considerably better, we're missing check on exception message. Is there a simple way to do it?